### PR TITLE
osd: cleanup multi-condition asserts

### PR DESCRIPTION
**Pull request by bridg8et**
fixes: https://trello.com/c/Ri6ixQ5X/615-osd-cleanup-multi-condition-asserts
signed-off-by: Bridget Rios <bridg8et94@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ideepika** [comment]
hey @bridg8et ! 

You might want to rebase this PR: 

you can do that by: 
1. git rebase -i HEAD~2 
2. you will be seeing 2 commits in the editors, please use drop instead of pick in the second and pick to reword and save it.
3. fix the Signed-off-by: Bridget Rios <bridg8et94@gmail.com>

File: src/erasure-code/shec/ErasureCodeShec.cc [hunk 1]
```diff
@@ -414,7 +414,7 @@ void ErasureCodeShecReedSolomonVandermonde::prepare()
   dout(10) << " [ technique ] = " <<
     ((technique == MULTIPLE) ? "multiple" : "single") << dendl;
 
-  ceph_assert((technique == SINGLE) || (technique == MULTIPLE));
+  ceph_assert((technique == SINGLE)) || ceph_assert((technique == MULTIPLE));
```
Comment (ideepika) [inline src/erasure-code/shec/ErasureCodeShec.cc:417]: @bridg8et changes looks good overall, just a small change: many places we have not needed double quotes, can you work on those instances eg: ceph_assert((technique == SINGLE))  -> ceph_assert(technique == SINGLE)

**Comment by ideepika** [comment]
@bridg8et are you working on this?

**Comment by ideepika** [comment]
@bridg8et I think this PR might take some more time, if it is hampering with OSD contribution requirement, you can always pick any other issue : from here https://tracker.ceph.com/issues/51379 that have not been taken https://pad.ceph.com/p/open-source-day let me know if you need assistance

File: src/krbd.cc [hunk 2]
```diff
@@ -383,8 +383,8 @@ class UdevMapHandler {
           *m_minnum = udev_device_get_property_value(p.get(), "MINOR");
           ceph_assert(*m_majnum == udev_device_get_sysattr_value(
                           m_bus_dev.get(), "major"));
-          ceph_assert(!have_minor_attr() ||
-                      *m_minnum == udev_device_get_sysattr_value(
+          ceph_assert(!have_minor_attr()) ||
+                     ceph_assert(*m_minnum == udev_device_get_sysattr_value(
```
Comment (idryomov) [inline src/krbd.cc:387]: This doesn't compile (and if it somehow compiled, it would introduce a bug as it is not an equivalent transformation).

**Review by idryomov — CHANGES_REQUESTED** [review]

**Comment by idryomov** [comment]
Closing as it is quite obvious that this change hasn't even been compiled before submission and there seems to be a concern with regard to significantly increasing the number of asserts in the code base.

**Pull request closed without merging**